ADO.NET/OAuthSampleApp/OAuthSampleApp.cs

```diff
@@ -0,0 +1,217 @@
+﻿using Vertica.Data.VerticaClient;
+using Newtonsoft.Json.Linq;
+using System.Configuration;
+using System.Runtime.InteropServices;
+using System.ComponentModel;
+
+internal class OAuthSampleApp
+{
+    private static VerticaConnectionStringBuilder connectionStringBuilder;
+    private static VerticaConnection vConnection;
+    private const string OAUTH_ACCESS_TOKEN_VAR_STRING = "OAUTH_SAMPLE_ACCESS_TOKEN";
+    private const string OAUTH_REFRESH_TOKEN_VAR_STRING = "OAUTH_SAMPLE_REFRESH_TOKEN";
+
+    static async Task Main(string[] args)
+    {
+        connectionStringBuilder = new VerticaConnectionStringBuilder();
+        connectionStringBuilder.ConnectionString = ConfigurationManager.ConnectionStrings["ConnectionString"].ConnectionString;
+
+        await SetUp();
+        await EnsureAccessToken();
+        await ConnectToDatabase();
+        await TearDown();
+    }
+
+    // admin must log in and setup the user for OAuth
+    // steps can also be followed in vsql
+    static async Task SetUp()
+    {
+        vConnection = new VerticaConnection(connectionStringBuilder.ConnectionString);
+        vConnection.Open();
+
+        string USER = ConfigurationManager.AppSettings["User"];
+        string CLIENT_ID = ConfigurationManager.AppSettings["ClientId"];
+        string CLIENT_SECRET = ConfigurationManager.AppSettings["ClientSecret"];
+        string INTROSPECT_URL = ConfigurationManager.AppSettings["TokenUrl"] + "introspect";
+
+        VerticaCommand cmd = new VerticaCommand("DROP USER IF EXISTS " + USER + " CASCADE", vConnection);

```

Comment:
Could we do something like this to avoid tons of cmd.ExecuteNonQuery(); lines. This seems a bit cleaner
```
string[] sqlCommands = {
    "DROP USER IF EXISTS " + USER + " CASCADE",
    "DROP AUTHENTICATION IF EXISTS adooauth CASCADE",
    "CREATE USER " + USER,
    "CREATE AUTHENTICATION adooauth METHOD 'oauth'" + GetAuthHost(),
    "GRANT AUTHENTICATION adooauth TO " + USER,
    "GRANT ALL ON SCHEMA PUBLIC TO " + USER,
    "ALTER AUTHENTICATION adooauth SET client_id = '" + CLIENT_ID + "'",
    "ALTER AUTHENTICATION adooauth SET client_secret = '" + CLIENT_SECRET + "'",
    "ALTER AUTHENTICATION adooauth SET introspect_url = '" + INTROSPECT_URL + "'"
};

// Execute each SQL command in the array
foreach (string commandText in sqlCommands) {
    VerticaCommand cmd = new VerticaCommand(commandText, vConnection))
    cmd.ExecuteNonQuery();
}
```

Reply:
Sure, that looks better.